src/spec_swagger/swagger2/core.cljc

```diff
@@ -1,6 +1,94 @@
 (ns spec-swagger.swagger2.core
   (:require [clojure.walk :as walk]
-            [spec-swagger.swagger2.schema :as schema]))
+            [clojure.spec.alpha :as s]
+            [spec-tools.json-schema :as json-schema]
+            [spec-tools.visitor :as visitor]
+            [spec-tools.core :as st]))
+
+;;
+;; conversion
+;;
+
+(defn- spec-dispatch [dispatch _ _ _] dispatch)
+(defmulti accept-spec spec-dispatch :default ::default)
+
+(defmethod accept-spec 'clojure.core/float? [_ _ _ _] {:type "number" :format "float"})
+(defmethod accept-spec 'clojure.core/double? [_ _ _ _] {:type "number" :format "double"})
+(defmethod accept-spec 'clojure.core/nil? [_ _ _ _] {})
+
+(defmethod accept-spec 'clojure.spec.alpha/or [_ _ children _]
+  ;; :anyOf is not supported by Swagger 2.0, so we just the the first child. In principle,
+  ;; we could do better in some special cases. For example, a reasonable schema
+  ;; for (s/or ::int int? ::str string?) would be {:type ["number", "string"]}.
+  (assoc
+    (first children)
+    :x-anyOf children))
+
+;; allOf is not supported
+(defmethod accept-spec 'clojure.spec.alpha/and [_ _ children _]
+  (assoc
+    (first children)
+    :x-allOf children))
+
+;; anyOf is not supported
+(defmethod accept-spec 'clojure.spec.alpha/alt [_ _ children _]
+  (assoc
+    (first children)
+    :x-anyOf children))
+
+;; anyOf is not supported
+(defmethod accept-spec 'clojure.spec.alpha/cat [_ _ children _]
+  {:type "array"
+   :minItems (count children)
+   :maxItems (count children)
```

Comment:
I'm now thinking that these counts are in fact wrong, because inside cat you can use regular expressions.